commands/proxy.ts

```diff
@@ -0,0 +1,103 @@
+import * as commandParams from "../command-params";
+import { EOL } from "os";
+
+export class ProxySetCommand implements ICommand {
+	public disableAnalytics = true;
+
+	public allowedParameters = [new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector)];
+
+	constructor(private $injector: IInjector,
+		private $logger: ILogger,
+		private $prompter: IPrompter,
+		private $proxyService: IProxyService) {
+	}
+
+	public async execute(args: string[]): Promise<void> {
+		let hostname = args[0],
+			port = args[1],
+			username = args[2],
+			password = args[3];
+
+		if (!hostname) {
+			hostname = await this.$prompter.getString("Hostname", { allowEmpty: false });
+		}
+
+		if (!port) {
+			port = await this.$prompter.getString("Port", { allowEmpty: false });
+		}
+
+		if (!username) {
+			username = await this.$prompter.getString("Username", { defaultAction: () => "" });
+		}
+
+		if (username && !password) {
+			password = await this.$prompter.getPassword("Password");
+		}
+
+		if (username && password) {
+			await this.$proxyService.setCredentials({
+				password,
+				username
+			});
+		}
+
+		const proxyCache: IProxyCache = {
+			PROXY_HOSTNAME: hostname,
+			PROXY_PORT: +port,
+			USE_PROXY: true
+		};
+
+		this.$proxyService.setCache(proxyCache);
+		this.$logger.out("Sucessfully setup proxy.");
```

Comment:
The command requires disabling of analytics, however I would like to know how many of the customers are using it. Can you add some feature tracking here. You can follow the syntax for commands or introduce new `feature` in the analytics.